rust/src/sys.rs

```diff
@@ -0,0 +1,21 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// We do this as an include so we can allow non_camel_case_types.
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+include!("_sys.rs");
```

Comment:
After talking to @catenacyber I  think I'll move the sys stuff to the more idiomatic Rust use of a `_sys` crate.  So `suricata_sys` will be a crate that only contains C bindings, and not depend on the `suricata` crate itself.

I guess it would be nice if it could contain the "C" functions written in Rust as well.